### PR TITLE
fix `kratos new -r` not support scp styled git url.issue https://gith…

### PR DESCRIPTION
**Pull request by czyt**
ref https://github.com/golang/go/blob/master/src/cmd/go/internal/vcs/vcs.go
see https://go-review.googlesource.com/c/go/+/12226/
git url define https://git-scm.com/docs/git-clone#_git_urls

### CODE REVIEW

**Comment by shenqidebaozi** [comment]
please fix lint

**Comment by czyt** [comment]
@shenqidebaozi lint fixed

File: cmd/kratos/internal/base/repo_test.go [hunk 1]
```diff
@@ -1,33 +1,44 @@
...
 func TestRepo(t *testing.T) {
-	os.RemoveAll("/tmp/test_repo")
-	r := NewRepo("https://github.com/go-kratos/service-layout.git", "")
-	if err := r.Clone(context.Background()); err != nil {
-		t.Fatal(err)
-	}
-	if err := r.CopyTo(context.Background(), "/tmp/test_repo", "github.com/go-kratos/kratos-layout", nil); err != nil {
-		t.Fatal(err)
-	}
+	//os.RemoveAll("/tmp/test_repo")
```
Comment (Windfarer) [inline cmd/kratos/internal/base/repo_test.go:8]: if these codes are unused, remove them

File: cmd/kratos/internal/base/vcs_url.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+	// Iterate over insecure schemes too, because this function simply
+	// reports the state of the repo. If we can't see insecure schemes then
+	// we can't report the actual repo URL.
+	for _, s := range scheme {
+		if repoURL.Scheme == s {
+			return repoURL, nil
+		}
+	}
+	return nil, errors.New("unable to parse repo url ")
```
Comment (Windfarer) [inline cmd/kratos/internal/base/vcs_url.go:57]: remove trailing space

**Comment by czyt** [comment]
@Windfarer hello,code issues fixed,pls review 😉

**Review by tonybase — APPROVED** [review]

**Review by haiyux — APPROVED** [review]

**Pull request merged**